index.js

```diff
@@ -14,4 +14,91 @@ module.exports = (robot, _, Settings = require('./lib/settings')) => {
+      const owner = result.data.owner.login
+      const repoName = result.data.name
+      // As context.repo() was undefined so had to convert it into object
+      const repo = {
+        owner: owner,
+        repo: repoName
+      }
+      // repo should have a .github folder
+      const path = '.github'
+      try {
+        const repoInfo = await context.github.repos.getContent({owner: owner, repo: repoName, path: path})
+
+        const FILE_NAME = repoInfo.data.find(async file => {
+          if (file.name === 'settings.yml') {
+            return file.name
+          } else {
+            try {
+              const reference = await context.github.gitdata.getReference({owner: owner, repo: repoName, ref: 'heads/master'})
+              const refData = reference.data
+              const sha = refData.object.sha
+              try {
+                const getRepo = await context.github.repos.get({owner: owner, repo: repoName})
+                await context.github.gitdata.createReference({owner: owner, repo: repoName, ref: 'refs/heads/probot', sha: sha})
+                // setting the template of file
+                const string = 'repository: \n' +
+                              '  name: ' + getRepo.data.name + ' \n' +
+                              '  description: ' + getRepo.data.description + ' \n' +
+                              '  homepage: ' + getRepo.data.homepage + ' \n' +
+                              '  topics: github, probot \n' +
+                              '  private: ' + getRepo.data.private + ' \n' +
+                              '  has_issues: ' + getRepo.data.has_issues + ' \n' +
+                              '  has_projects: ' + getRepo.data.has_projects + ' \n' +
+                              '  has_wiki: ' + getRepo.data.has_wiki + ' \n' +
+                              '  has_downloads: ' + getRepo.data.has_downloads + ' \n' +
+                              '  default_branch: ' + getRepo.data.default_branch + ' \n' +
+                              '  allow_squash_merge: ' + getRepo.data.allow_squash_merge + ' \n' +
+                              '  allow_merge_commit: ' + getRepo.data.allow_merge_commit + ' \n' +
+                              '  allow_rebase_merge: ' + getRepo.data.allow_rebase_merge + ' \n\n' +
+                              'labels: \n' +
+                              '  - name: bug \n' +
+                              '    color: CC0000 \n' +
+                              '  - name: feature \n' +
+                              '    color: 336699 \n' +
+                              '  - name: first-timers-only \n' +
+                              '    oldname: bug \n\n' +
+                              'collaborators: \n' +
+                              '  - username: enter any username \n' +
+                              '    permission: push'
+
+                const base64 = require('./encode')
+                const encodedString = base64.encode(string)
```

Comment:
@knrt10 instead of using some custom created Base64 encoding, you can [use Node.JS's built-in `Buffer` global](https://stackoverflow.com/questions/6182315/how-to-do-base64-encoding-in-node-js):

```js
const string = 'Hello World'
const base64 = Buffer.from(string).toString('base64')
```

Reply:
Ok let me update both the changes

Reply:
@knrt10 i was asking about this only.

Reply:
@aryannarora finally changed it.